### PR TITLE
Add changes for edge-21.4.2

### PR DESCRIPTION
**Pull request by mateiidavid**
## edge-21.4.2

This edge release is another candidate for `stable-2.10.1`!

It includes some CLI fixes and addresses an issue where the outbound proxy
could use an incorrect client for the request when running in ingress mode.

Thank you to all of our users that have helped test and identify issues in 2.10!

* Fixed an issue where the CLI would add the wrong annotation when using the
  `--ingress` flag
* Fixed a nil pointer dereference in `linkerd repair` caused by a mismatch
  between CLI and server versions
* Removed an unnecessary error handling condition in multicluster check
  (thanks @wangchenglong01!)
* Fixed a caching issue where incorrect clients could be used for requests
  when outbound proxy runs in ingress mode
* Removed unsupported `matches` field from TrafficSplit CRD

Signed-off-by: Matei David <matei@buoyant.io>

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,5 +1,23 @@
...
+
+This edge release is another candidate for `stable-2.10.1`!
+
+It includes some CLI fixes and addresses an issue where the outbound proxy
+could use an incorrect client for the request when running in ingress mode.
+
+Thank you to all of our users that have helped test and identify issues in 2.10!
+
+* Fixed an issue where the CLI would add the wrong annotation when using the
+  `--ingress` flag
```
Comment (alpeb) [inline CHANGES.md:13]: Can you clarify this is about the `linkerd inject` command?

**Review by alpeb — APPROVED** [review]
LGTM modulo nit above :+1:

File: CHANGES.md [hunk 2]
```diff
@@ -1,5 +1,23 @@
...
+* Fixed a nil pointer dereference in `linkerd repair` caused by a mismatch
+  between CLI and server versions
+* Removed an unnecessary error handling condition in multicluster check
+  (thanks @wangchenglong01!)
+* Fixed a caching issue where incorrect clients could be used for requests
```
Comment (adleong) [inline CHANGES.md:18]: Can we reword this to frame it around the user impact? Is this #5975?

**Review by adleong — APPROVED** [review]

**Review by alpeb — APPROVED** [review]

**Review by olix0r — APPROVED** [review]

**Review by adleong — APPROVED** [review]
